### ISSUE TITLE
usecase: Cohosting from a cron job

### ISSUE THREAD

**Issue by olizilla** [ipfs-shipyard/ipfs-cohost#14]
I want to pass a list of sites to ipfs-cohost that may change over time. I only want to keep the most recent version of the site cohosted, and I want new sites to be added and old sites to be removed. From cron, i'd call something like:

```console
$ ipfs-cohost website.ipfs.io website.filecoin.io ipld.io shortlived.io
```

and all i want to see in my cohosting dir is the list of sites, fully cohosted, with only the most recent snapshot. Then a week later we add site and remove one we no longer need:

```diff
- ipfs-cohost website.ipfs.io website.filecoin.io ipld.io shortlived.io
+ ipfs-cohost website.ipfs.io website.filecoin.io ipld.io newthing.eth
```

and have `ipfs-cohost` figure out that we want to add a snapshot for `newthing.eth` and remove `shortlived.io`

in an infra-as-code set up you'd keep the list of domains in git, and would want to be able to call the same command over and over with that list, and have it do the right thing when it changed. 

The mode where i have to pass a list to a command once, and then know that i have to call sync, or add or remove is much harder to implement in an automated set up.

**Comment by hacdias** [comment]
@olizilla so what you want is:

1. Pass all websites to be cohosted;
2. Only last snapshot for each;
3. Only cohost passed websites.

Maybe we could have a subcommand for that or flag, such as:

```
ipfs-cohost example.com example.org --strict --single-snapshot
```

---

I don't feel comfortable with the world `strict`.

We could also call it as a subcommand such as `mirror` where it would just mirror the current snapshots of each website and only cohost those!

```
ipfs-cohost mirror example.com example.org

---
```

I actually would love something like this personally too.

/cc @lidel

**Comment by lidel** [comment]
This sounds like a new command, and `latest` may be a good name for it:

```
ipfs-cohost latest example.com example.org
```

(`latest` is basically a shortcut for multiple `add` + `purge 1` + `rm` if not on the passed list)

**Comment by hacdias** [comment]
@lidel yes, that was my second suggestion and let's go for it. I would just make it after #16 so this sub-command would just be `rm --all + add`.

**Comment by hacdias** [comment]
However, if we add `rm --all`, then we just need to call two commands, which is not a big deal. Wdyt?

**Comment by lidel** [comment]
I think adding `only-latest` as an atomic alias would make cron and one-off use cases much easier. 

@olizilla any preferences on name/sempantics?

**Comment by hacdias** [comment]
@lidel yes, I did it #20
